app/unit-tests/src/org/commcare/android/CommCareTestRunner.java

```diff
@@ -33,6 +34,7 @@ public InstrumentationConfiguration createClassLoaderConfig(final FrameworkMetho
         InstrumentationConfiguration.Builder builder = new InstrumentationConfiguration.Builder(super.createClassLoaderConfig(method));
         builder.addInstrumentedPackage("net.sqlcipher.database.SQLiteDatabase");
         builder.addInstrumentedPackage("org.commcare.models.encryption");
+        builder.addInstrumentedClass(ConnectionDiagnosticTask.class.getName());
```

Comment:
Is just shadowing with the `CommcareDiagnosticTaskMock` class  not enough ?

Reply:
Hey, I'm not sure from where I got this, but it seems like the Shadow class won't be invoked while testing unless I add it to the build runner. The best possible reason I can find is [here](https://stackoverflow.com/a/29641926/6671572) that in order to shadow our own classes(originally we can only shadow android classes) we need to add the class in the custom test runner.